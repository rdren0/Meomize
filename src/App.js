import React, { Component } from 'react';
import logo from './logo.svg';
import Review from './Components/Review';
import Header from './Components/Header';


import './App.css';

export class App extends Component {
  constructor() {
    super();
    this.state = {
      cards: []
    }

    }

    // componentDidMount() {
    // fetch('http://whateverly-datasets.herokuapp.com/api/v1/gems')
    //   .then(response => response.json())
    //   .then(cards => {
    //     this.setState({
    //       cards: cards.cards
    //     })
    //   })
    //   .catch(err => {
    //     throw new Error(err);
    //   })
    // }
  


  render() {
    return (
      <div className="App">
      <Header />
      <Review />
      </div>
    );
  }
}

export default App;
