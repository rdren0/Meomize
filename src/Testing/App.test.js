import React from 'react';
import App from '../App';
import { shallow } from 'enzyme';
import {configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import ReactDOM from 'react-dom';

configure({ adapter: new Adapter() });


const mockInitalState = {
      cards: [],
      loadingScreen: true
  }
describe('App initial state', () => {
  let wrapper;
  wrapper = shallow(<App />);
   it('should match the snapshot with initial state', () => {
    expect(wrapper).toMatchSnapshot();
  });
   it('should have the correct default state', () => {
    expect(wrapper.state()).toEqual(mockInitalState)
   })
})


describe('App', () => {
  let wrapper;

  beforeEach(() => {
   wrapper = shallow(<App />);
   wrapper.setState({ cards: [],
      loadingScreen: true})
 })

  it('should match the snapshot with mock state', () => {
    expect(wrapper).toMatchSnapshot();
  });


  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });


});
