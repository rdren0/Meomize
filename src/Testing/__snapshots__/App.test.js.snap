// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App initial state should match the snapshot with initial state 1`] = `
<div>
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
  <p>
    Loading Data...
  </p>
</div>
`;

exports[`App should match the snapshot with mock state 1`] = `
<div>
  <img
    alt="logo"
    className="App-logo"
    src="logo.svg"
  />
  <p>
    Loading Data...
  </p>
</div>
`;
